workshops/pacman game/README.md

```diff
@@ -0,0 +1,337 @@
+                vector(-5, 0),
+                vector(0, 5),
+                vector(0, -5),
+            ]
+            plan = choice(options)
+            course.x = plan.x
+            course.y = plan.y
+
+        up()
+        goto(point.x + 10, point.y + 10)
+        dot(20, 'red')
+
+    update()
+
+    for point, course in ghosts:
+        if abs(pacman - point) < 20:
+            return
+    ontimer(move, 100)
+```
+
+* **NOTE:** Our Pac-man has no sense of direction yet. It's using the default `aim` we assigned earlier when we were initializing variables. At the moment, our Pac-man is only programmed to go right, at a speed of 5. We'll add some direction in the next step.
+* The `writer` variable we implemented at the beginning is being used here with the `undo()` and `write()` functions. These are built-in functions that come with `Turtle`. 
+* As we go down further the code, we're checking if the movement is `valid`. Remember those physics functions we made? This is where we're implementing it. A move is `valid` when it doesn't involve touching a brick.
+* We're also utilizing the `ghosts` array with vectors we initialized before. This lets us have them in the game and move them around.
+* We're creating `dots` (a function from `Turtle`) that our Pac-man will eat. We are ensuring they're being created dynamically as they are attached to the Pac-man's moves. If the Pac-man doesn't move, it won't spawn more dots.
+* **Going a step further: **If you want to take things a step further and introduce an animation, sound effect, or other prettiness factor, we would introduce it after `brick(x, y)`. Try adding `print(“PAC-MAN”)` as an example.
+
+## Giving Direction to Pac-man
+
+We created a function that lets our Pac-man move, but it doesn't know where to go. It only followed the direction that we provided to it in the `aim` variable. Let's create a function that lets our Pac-man change that `aim` value, or in other terms, change it's direction. Add this code into the bottom of your code:
+
+```python
+def change(x, y):
+    if valid(pacman + vector(x, y)):
+        aim.x = x
+        aim.y = y
+```
+
+We first check if the direction is valid (we aren't inside of a brick or anything), then we set the `aim` we initialized at the start of the code.
+
+## The Final Stretch!
+
+As we mentioned earlier, this game is coded using a principle called modular programming. This means that everything we code is in functions, now we can bring it together. We can call these functions by writing the function name and the parameters it needs in parentheses. This code block will put all the pieces together:
+
+```python
+setup(420, 420, 370, 0)
+hideturtle()
+tracer(False)
+writer.goto(160, 160)
+writer.color('white')
+writer.write(state['score'])
+listen()
+onkey(lambda: change(5, 0), 'Right')
+onkey(lambda: change(-5, 0), 'Left')
+onkey(lambda: change(0, 5), 'Up')
+onkey(lambda: change(0, -5), 'Down')
+world()
+move()
+done()
+```
+
```

Comment:
Maybe here add something like "Congrats! You've just made your own Pacman game! 🎉"